### ISSUE TITLE
Content event's clickStartShape/clickEndShape not correctly reset 

### ISSUE THREAD

**Issue by albertogasparin** [konvajs/konva#318]
On `contentMousedown` Konva sets `clickStartShape` and on `contentMouseup` it sets `clickEndShape`, however that value never gets set back to `null` in case the event is fired on an empty area afterwards.

https://github.com/konvajs/konva/blob/master/src/Stage.js#L513
```js
// ...
var shape = this.getIntersection(this.getPointerPosition());

Konva.listenClickTap = true;

this.clickStartShape = shape; // <- proposed fix
if (shape && shape.isListening()) {
    // this.clickStartShape = shape; // <- current code
    shape._fireAndBubble(MOUSEDOWN, { evt: evt });
} else {
// ...
```

Same on https://github.com/konvajs/konva/blob/master/src/Stage.js#L561
Happy to make a PR if you agree with the solution.

**Comment by albertogasparin** [comment]
In my case yes, as I'm trying to detect when I'm interacting with a shape. 
Let's assume I'm implementing a select/deselect behavior. I have a large stage with a small shape inside it. When I click on a shape, `contentMousedown` is fired with a correct `clickStartShape` so I can trigger the selection. However, if the user later clicks on an empty area of the stage, the event fires with the old `clickStartShape` and triggers again the selection instead of the deselect behavior.

**Comment by lavrton** [comment]
You should not use `clickStartShape`. This is private property.
I slightly change the behaviour of some event for a stage.
You can do something like this:
```js
stage.on('click', (e) => {
  if (e.target === stage) {
     console.log('click on empty area');
     // deselect code
  } else {
    console.log('click on shape', e.target);
    // select code
  }
});

**Comment by lavrton** [comment]
These changes are not published yet. But already on the master branch.

**Comment by albertogasparin** [comment]
That's good news. Glad you properly implemented it on the event object. 
I agree that `clickStart`/`clickEnd` were probably not supposed to be used but currently they are the only way I've found to do what I need without implementing it by myself. 
So, if you are not interesting into changing that code, feel free to close this issue.

BTW, if someone else has my same issue till the new version gets released, I've fixed it with a hack on my end:

```js 
myStage.on('contentMouseup', function (ev) {
  setTimeout(function () {
    myStage.clickStartShape = null;
    myStage.clickEndShape = null;
  }, Konva.dblClickWindow + 20);
});
```
